creator-node/test/nodesync.test.js

```diff
@@ -1,14 +1,348 @@
+      const appInfo = await getApp(ipfs, libsMock, blacklistManager)
+      server = appInfo.server
+      app = appInfo.app
+    })
+
+    it('should export the correct json blob for db state with a track and user', async function () {
+      // SETUP
+
+      const MAX_CLOCK_VAL = 37
+
+      // Create user
+      const { cnodeUserUUID, sessionToken } = await createStarterCNodeUser()
+      await getCNodeUser(cnodeUserUUID)
+
+      // Set user metadata
+      const metadata = {
+        metadata: {
+          testField: 'testValue'
+        }
+      }
+      const { body: { data: { metadataMultihash, metadataFileUUID } } } = await request(app)
+        .post('/audius_users/metadata')
+        .set('X-Session-ID', sessionToken)
+        .send(metadata)
+
+      const associateRequest = {
+        blockchainUserId: 1,
+        metadataFileUUID,
+        blockNumber: 10
+      }
+
+      // Associate user with metadata
+      await request(app)
+        .post('/audius_users/')
+        .set('X-Session-ID', sessionToken)
+        .send(associateRequest)
+
+      // Upload a track
+      const file = fs.readFileSync(testAudioFilePath)
+      // set track content
+      const { body: { transcodedTrackCID, transcodedTrackUUID, track_segments: trackSegments, source_file: sourceFile } } = await request(app)
+        .post('/track_content')
+        .attach('file', file, { filename: 'fname.mp3' })
+        .set('Content-Type', 'multipart/form-data')
+        .set('X-Session-ID', sessionToken)
+
+      // set track metadata
+      const trackMetadata = {
+        test: 'field1',
+        owner_id: 1,
+        track_segments: trackSegments
+      }
+      const { body: { metadataMultihash: trackMetadataMultihash, metadataFileUUID: trackMetadataFileUUID } } = await request(app)
+        .post('/tracks/metadata')
+        .set('X-Session-ID', sessionToken)
+        .send({ metadata: trackMetadata, source_file: sourceFile })
+      // associate track metadata with track
+      await request(app)
+        .post('/tracks')
+        .set('X-Session-ID', sessionToken)
+        .send({
+          blockchainTrackId: 1,
+          blockNumber: 10,
+          metadataFileUUID: trackMetadataFileUUID,
+          transcodedTrackUUID
+        })
+
+      // Test: export
+
+      const { pubKey } = testEthereumConstants
+      const { body: exportBody } = await request(app)
+        .get(`/export?wallet_public_key=${pubKey.toLowerCase()}`)
+
+      // Verify
+
+      // Get user metadata
+      const userMetadataFile = stringifiedDateFields((await models.File.findOne({
+        where: {
+          multihash: metadataMultihash,
+          fileUUID: metadataFileUUID,
+          clock: 1
+        },
+        raw: true
+      })))
+
+      // get transcoded track file
+      const copy320 = stringifiedDateFields((await models.File.findOne({
+        where: {
+          multihash: transcodedTrackCID,
+          fileUUID: transcodedTrackUUID,
+          clock: 3
```

Comment:
technically i don't think the clock values for the 320kpbs and segmented files are deterministic because it ahppens inside a Promise.all(), but i think for now this is probably fine. the better way to do this would be to use the 'type' column as a filter to query for type 'copy320' or 'file'

Reply:
fixed